inc/saivlan.h

```diff
@@ -98,6 +98,12 @@ typedef enum _sai_vlan_attr_t
       * Value Range SAI_SWITCH_ATTR_VLAN_USER_META_DATA_RANGE */
     SAI_VLAN_ATTR_META_DATA,
 
+    /**
+      * The Multicast 802.1BR E-Channel Id (ECID), to be used in the flooding traffic
+      * sent by the CB to Port Extenders (PE). [sai_uint32_t] (CREATE_AND_SET) (default to 0)
+      */
+    SAI_VLAN_ATTR_FLOODING_ECID,
```

Comment:
Dell - Internal Use - Confidential
Hi Min Yao,

The ECID is global in a dot1br domain that contains a single CB and a set of associated PEs. I am not clear on what you mean by “Internal External Bridge”. Do you mean to refer to the CB domain?

A Vlan can have both non dot1br ports (native ports) and Extended ports (remote PE port). So unknown L2 traffic must be sent to both types of ports. Unknown unicast/flooding in a vlan can be achieved by any of the following mechanisms:


1.      CB Replication

CB can replicate a copy for each of the Extended ports as unicast frames (the ECID field will be set to the E-channel id of the extended port. It must ensure that source suppression is achieved here

2.      CB and PE Replication

CB can replicate a copy for each PE. PE would then replicate copies to its dot1br access ports. If there are 4 extended ports, 2 connected to PE1 and other 2 connected to PE2, then CB will send 1 copy to PE1 and another copy to PE2. The Ecid will be the same in both these copies. The PEs will then replicate the packet.

In this scenario the frame must carry the multicast ECID. This ECID is nothing bug SAI_VLAN_ATTR_FLOODING_ECID. If this attribute is not present, how will CB determine the multicast ECID that must be filled in the flooding traffic?

I am not clear with the following statement,

“A packet could be flooded to normal access ports and multiple multicast E-Channels at the same time.”

Can you please elaborate on the scenario, where there is a need for Vlan flooding (unknown L2 traffic on the vlan), to be sent on multiple multicast E-Channels? As per my understanding the unknown L2 traffic on the vlan has to be sent only on one multicast E-Channel. Only L3 Multicast would require multiple multicast E-Channels? Are you referring to L3 Multicast traffic?

Regards,
Ravi

From: Min Yao [mailto:notifications@github.com]
Sent: Tuesday, August 30, 2016 1:42 PM
To: opencomputeproject/SAI <SAI@noreply.github.com>
Cc: Sivasankar, Ravikumar <Ravikumar_Sivasankar@Dell.com>; Mention <mention@noreply.github.com>
Subject: Re: [opencomputeproject/SAI] IEEE 802.1BR SAI proposal. (#73)


In inc/saivlan.h<https://github.com/opencomputeproject/SAI/pull/73#discussion_r76748108>:

> @@ -98,6 +98,12 @@ typedef enum _sai_vlan_attr_t

>        * Value Range SAI_SWITCH_ATTR_VLAN_USER_META_DATA_RANGE */

>      SAI_VLAN_ATTR_META_DATA,

>

> +    /**

> +      * The Multicast 802.1BR E-Channel Id (ECID), to be used in the flooding traffic

> +      * sent by the CB to Port Extenders (PE). [sai_uint32_t] (CREATE_AND_SET) (default to 0)

> +      */

> +    SAI_VLAN_ATTR_FLOODING_ECID,

@rsivasankar<https://github.com/rsivasankar> ECID is not a global concept, it is allocated within the scope of an Internal External Bridge. There could be multiple Internal External Bridges within a CB. So ECID can not be used globally. And I don't understand the purpose of this FLOODING_ECID. If it's for PE replication (multicast ECID), then it's wrong. A packet could be flooded to normal access ports and multiple multicast E-Channels at the same time. For CB replication, this attribute is not needed.

PS; Suggest adding new switch level attribute to indicate that the switch is a CB or PE.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/opencomputeproject/SAI/pull/73/files/279f0335c33c0c831ec955e1cf93daf61adf52d5#r76748108>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AOoHu-7D1igwCvDxhExGbxUkAsjn1yeRks5qk-XOgaJpZM4HpSGE>.